### PR TITLE
HDPI-6415 IDAM login navigation

### PR DESCRIPTION
**Pull request by libanAbdirahman1**
### Jira link

[HDPI-6415](https://tools.hmcts.net/jira/browse/HDPI-6415)

### Change description

- Adding claim list route to display claims a user is linked to as a defendant. They are redirected to this screen upon login.

### Testing done

- Tests added for new route and existing tests updated
Screenshot 2026-06-01 at 16 34 59
Screenshot 2026-06-01 at 16 30 24

### Security Vulnerability Assessment ###

**CVE Suppression:** Are there any CVEs present in the codebase (either newly introduced or pre-existing) that are being intentionally suppressed or ignored by this commit?
  * [ ] Yes
  * [x] No

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/routes/claimList.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    await req.i18n?.loadNamespaces(['claimList']);
+    const t = getTranslationFunction(req, ['claimList', 'common']);
+
+    try {
+      const claimsAgainst = await getCitizenClaims(accessToken);
+      const tableRows = claimsAgainst.map(claim => [
+        { text: claim.caseRef },
+        { text: claim.claimantName },
```
Comment (andrewdami11) [inline src/main/routes/claimList.ts]: May we need to validate `claim.caseRef` before using it in the link HTML, given `caseRef` is optional in `ClaimSummary`? Would filtering invalid rows (or rendering a non-link fallback) avoid `/case/undefined/dashboard`?

File: src/test/unit/routes/claimList.test.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    expect(res.redirect).toHaveBeenCalledWith('/login');
+    expect(res.render).not.toHaveBeenCalled();
+  });
+
+  it('renders claimList with mapped table rows on successful fetch', async () => {
+    (getCitizenClaims as jest.Mock).mockResolvedValue([
+      { caseRef: '1234567890123456', claimantName: 'John Doe', propertyPostcode: 'SW1A 1AA' },
+      { caseRef: '9876543210987654', claimantName: 'Jane Smith', propertyPostcode: 'EC1A 1BB' },
+    ]);
```
Comment (andrewdami11) [inline src/test/unit/routes/claimList.test.ts:81]: Should we add a test for an empty or partially invalid claims payload to assert the `noClaimsFound`/safe rendering path? That would help lock in behavior when `caseRef` or other fields are missing?

File: src/main/routes/claimList.ts (hunk 1 above)
Comment (libanAbdirahman1) [reply src/main/routes/claimList.ts]: i've actioned this [here](https://github.com/hmcts/pcs-frontend/pull/1444/commits/deb0460d9d15c7c0ed5357b31fc6d2ad6207d3a1)

File: src/test/unit/routes/claimList.test.ts (hunk 2 above)
Comment (libanAbdirahman1) [reply src/test/unit/routes/claimList.test.ts:81]: i've actioned this [here](https://github.com/hmcts/pcs-frontend/pull/1444/commits/deb0460d9d15c7c0ed5357b31fc6d2ad6207d3a1)

**Review by andrewdami11 — APPROVED** [review]

**Review by arun-hmcts — APPROVED** [review]
